textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -0,0 +1,142 @@
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.dataflow.IOperator;
+import edu.uci.ics.textdb.common.constants.SchemaConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.field.IntegerField;
+import edu.uci.ics.textdb.common.field.StringField;
+
+/**
+ * @author Sudeep [inkudo]
+ * 
+ */
+public class DictionaryMatcher implements IOperator {
+
+    private IOperator operator;
+    private IDictionary dict;
+    private String dictValue;
+    private int positionIndex; // next position in the field to be checked.
+    private int fieldIndex; // Index of the next field to be checked.
+    private int spanIndexVal; // Starting position of the matched dictionary
+                              // string
+    private ITuple spanTuple;
+    private List<IField> fields;
+
+    public DictionaryMatcher(IDictionary dict, IOperator operator) {
+        this.operator = operator;
+        this.dict = dict;
+    }
+
+    /**
+     * @about Opens dictionary matcher. Initializes positionIndex and fieldIndex
+     *        Gets first sourceoperator tuple and dictionary value.
+     */
+    @Override
+    public void open() throws Exception {
+        try {
+            positionIndex = 0;
+            fieldIndex = 0;
+            operator.open();
+            dictValue = dict.getNextDictValue();
+            spanTuple = operator.getNextTuple();
+            fields = spanTuple.getFields();
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * @about Gets next matched tuple. Returns a new span tuple including the
+     *        span results. Performs a scan based search, gets the dictionary
+     *        value and scans all the documents for matches
+     */
+    @Override
+    public ITuple getNextTuple() throws Exception {
+        if (fieldIndex < fields.size()) {
+            IField field = spanTuple.getField(fieldIndex);
+            if (field instanceof StringField) {
+                String fieldValue = ((StringField) field).getValue();
+
+                // Get position of dict value in the field.
+                if ((spanIndexVal = fieldValue.indexOf(dictValue, positionIndex)) != -1) {
+
+                    // Increment positionIndex so that next search occurs from
+                    // new positionIndex.
+                    positionIndex = spanIndexVal + fieldValue.length();
+
+                    Attribute attribute = spanTuple.getSchema().get(fieldIndex);
+                    String fieldName = attribute.getFieldName();
+
+                    //Creating a clone of the spanTuple and returning the cloned one,
+                    //so that the changes are not reflected in the original spanTuple.
+                    //The cloned SpanTuple is populated with the span related data.
+                    ITuple spanTupleCloned = spanTuple.clone();
```

Comment:
As said earlier, it seems better to create a new `SpanTuple` which will take the `tuple` and those attribute as the parameters of the constructor.